python/ipyleaflet/ipyleaflet/leaflet.py

```diff
@@ -1134,6 +1143,38 @@ def redraw(self):
         """
         self.send({"msg": "redraw"})
 
+    def set_feature_style(self, id:Int, layer_style:Dict):
+        """Re-symbolize one feature.
+
+        Given the unique ID for a vector features, re-symbolizes that feature across all tiles it appears in. 
+        Reverts the effects of a previous set_feature_style call. get_feature_id must be defined for 
+        set_feature_style to work. 
+
+        Attributes
+        ----------
+        id: int
+            The unique identifier for the feature to re-symbolize
+        layer_styles: dict
+            Style to apply to the feature
+        """
+        self.send({"msg": 
```

Comment:
I'm wondering if this could be a traitlets property instead of a custom comm message?

Something like 
```python
feature_styles = Dict().tag(sync=True)
```

`feature_styles` being the {id: layerStyle} dictionary?

Custom comm messages are transient and will not be saved in your widget model, resulting in many issues.

Reply:
How would that work in `VectorTileLayer.ts` so that the `this.obj.setFeatureStyle` is called when the python method is called?

Reply:
You would need to set a new value for `feature_styles` Python side:

```python
def set_feature_style(self, id:Int, layer_style:Dict):
    self.feature_styles = {**feature_styles, id: layer_style}
```

Then you would listen for changes in the JavaScript side:

```python
this.model.on("change:feature_styles", () => {
  const feature_styles = this.model.get("feature_styles");
  // You may need to reset all previous features styles here
  // [...]
  for (const id of Object.keys(feature_styles)) {
    this.obj.setFeatureStyle(id, feature_styles[id]);
  }
})
```

Reply:
Thanks, I'll give it a try and make a new commit. 

FYI I am also testing another commit with changes to how `rendererFactory`, `getFeatureId`, and `vectorTileLayerStyles` are exposed in python, because I encountered an issue as follows:

1. Create a vector tile layer with `renderer_factory='svg'`, which is a string. 
2. In javascript, the `rendererFactory` is set to a function `L.svg.tile`.
3. Change the opacity in python.
4. Move the map
5. Error in javascript -- because renderer_factory is in sync with `rendererFactory`, it's now set to the string `svg`, and it throws a javascript error that the `rendererFactory` is not a function.

The new change to avoid this issue is that in python, instead of `rendererFactory` I will have a separate option called `renderer`, which is used in javascript to create `rendererFactory`. A similar strategy can be used for `getFeatureId` and `vectorTileLayerStyles` because they are constructed in javascript as functions, they shouldn't be in 'sync' with python. 

I will aim to commit this change soon as well.